content/rdi/troubleshoot.md

```diff
@@ -0,0 +1,288 @@
+---
+Title: Troubleshoot and resolve common problems
+linkTitle: Troubleshooting
+description: Resolve common Redis Data integration problems
+weight: 50
+alwaysopen: false
+categories: ["redis-di"]
+aliases: 
+---
+
+Learn how to diagnose common problems during installation and runtime of Redis Data Integration.
+
+## Problems during installation
+
+### Problems with pip
+
+We have seen problems with using the [pip](https://pypi.org/project/pip/) installation due to the following reasons:
+
+- **Wrong Python 3 Version**
+
+  In many Linux distros the Python 3 version is still 3.6.\*. You can check this by running: `python --version`.
+
+- **More than one Python 3 Installed**
+
+  To solve this problem, create a Python virtual environment (venv) specifying
+  the path to the right Python 3 installation for example `<PATH>/python3.8 -m venv <VENV-NAME>`. If you are not sure about the path run `which python3` and copy the path pointing to a Python installation with a version greater than `3.6`.
+
+- **Outdated Pip Version**
+
+  Some old pip versions will have problems with the wheel filename. It is best to update pip as well. In you `venv` run `pip install --upgrade pip`.
+
+- **Old RDI CLI Libraries from Previous Installations**
+
+  Sometimes RDI CLI libraries or its dependencies are already installed and they need an upgrade to get the latest version:
+
+  ```bash
+  # outputs the list of installed packages
+  pip freeze > requirements.txt
+
+  # updates all packages
+  pip install -r requirements.txt --upgrade
+  ```
+
+### Problems with `create` command
+
+The following problems can occur when running the `create` command:
+
+- **No network access to Redis Enterprise API**
+  - Verify your CLI host can indeed resolve the DNS entry you use (if applicable) `dig <HOSTNAME>`.
+  - Verify you have access to the specified ip:port `telnet <IP ADDRESS OF SERVER PC> <PORT>`.
+- **Using a Redis Enterprise user with the wrong role**
+
+  User must have roles of either `DB Member`, `Cluster Member` or `Admin`.
+
+- **Trying to connect to the Redis Data Integration DB without an Ingress Controller path**
+
+  When running Redis Enterprise on Kubernetes, each Redis DB has a service that is created only after the DB had been created. In order to connect with the CLI from outside of Kubernetes and configure the Redis Data Integration DB, you need an `Ingress` resource for your Redis Data Integration database and `Ingress Controller` running on the Kubernetes cluster.
+
+  A Kubernetes `Service` and `Ingress` are mutually exclusive. So if your database has a `Service` you should remove it and add an `Ingress` resource. Read [Kubernetes docs](https://kubernetes.io/docs/concepts/services-networking/ingress/) for more information.
```

Comment:
Suggested a comma here. I'm also not sure about using "so" to start the second sentence. Maybe we can just remove it, replace it with another word, or rework these sentences a bit.
```suggestion
  A Kubernetes `Service` and `Ingress` are mutually exclusive. So if your database has a `Service`, you should remove it and add an `Ingress` resource. Read [Kubernetes docs](https://kubernetes.io/docs/concepts/services-networking/ingress/) for more information.
```